packages/grid/_modules_/grid/components/cell/GridCell.tsx

```diff
@@ -175,7 +180,7 @@ export const GridCell = React.memo(function GridCell(props: GridCellProps) {
       data-mode={cellMode}
       aria-colindex={colIndex + 1}
       style={style}
-      tabIndex={tabIndex}
+      tabIndex={cellMode === 'view' || !isEditable ? tabIndex : -1}
```

Comment:
Didn't we have an `enum` for the different cell modes? If not maybe we should have one.

Reply:
Yeah, I remember the discussion. The problem with adding an enum is that users won't be able to use plain strings. I tried to add an enum for the edit mode in https://github.com/mui-org/material-ui-x/pull/2098#discussion_r685855801 but we would force everyone to use the enum value, losing the simplicity.

Reply:
Hmm I'm not sure I understand. Having the `view` coming from a `enum` won't limit users to use simple strings to check: https://www.typescriptlang.org/play?#code/KYOwrgtgBAsg9gE2FA3gKClAagS2AdygF4oByANz31IBo0BfNNAYzhAGcAXKZ4AGz7wkxMpQKkA3E1Yc4fYADo+cAOYAKXgKHIiu2IkW4CASjRA

Or maybe I'm missing something?

Reply:
If we use the enum to type the argument, then passing a string won't work anymore.

https://www.typescriptlang.org/play?#code/KYOwrgtgBAsg9gE2FA3gKClAagS2AdygF4oByANz31IBo0BfNNAMzBAGMAXHOEKT4AGdOACgiJgALlgSAlKgxR2vQXAA2wAHRq4AcxGlmcOACMAhgCdSsgNwMmA4QcoFrdoA

Reply:
Maybe we can use a combination of a `type` and an `enum` then. I've done it like this for the `density` feature https://github.com/mui-org/material-ui-x/blob/master/packages/grid/_modules_/grid/models/gridDensity.ts

Reply:
I've created the enums `GridCellModes`, `GridRowModes` and `GridEditModes`. I'm not typing the argument with them, just using their value.